models/code.go

```diff
@@ -0,0 +1,24 @@
+package models
+
+// CodeResults contains an array of codes which can be paginated
+type CodeResults struct {
+	Count        int    `json:"count"`
+	Start        int    `json:"start_index"`
+	ItemsPerPage int    `json:"items_per_page"`
+	Items        []Code `json:"items"`
+	TotalCount   int    `json:"total_count"`
+}
+
+// Code for a single dimensions type
+type Code struct {
+	ID    string    `json:"-"`
+	Code  string    `json:"code"    bson:"code"`
```

Comment:
The Dataset API is defining a link to here which is {id: 1234, href: /codelist/{list_id}/codes/1234} - this suggests that what we're calling code here should be an ID? I appreciate that _id is needed from a mongo perspective, but I think it would be helpful if we called the Code here ID. Alternatively, we need to make sure the Dataset API is being provided a uuid from the codelist items here.

Reply:
I made code type to be returned as ID in JSON